explore_data/cluster_by_accessory_gene.py

```diff
@@ -0,0 +1,609 @@
+fig += pn.labs(x="PC 1", y="PC 2", title="RNA-seq expression using PA14 reference")
+fig += pn.theme_bw()
+fig += pn.theme(
+    legend_title_align="center",
+    plot_background=pn.element_rect(fill="white"),
+    legend_key=pn.element_rect(fill="white", colour="white"),
+    legend_title=pn.element_text(family="sans-serif", size=15),
+    legend_text=pn.element_text(family="sans-serif", size=12),
+    plot_title=pn.element_text(family="sans-serif", size=15),
+    axis_text=pn.element_text(family="sans-serif", size=12),
+    axis_title=pn.element_text(family="sans-serif", size=15),
+)
+fig += pn.guides(colour=pn.guide_legend(override_aes={"alpha": 1}))
+
+print(fig)
+
+# +
+# Plot
+fig = pn.ggplot(
+    pao1_pa14_core_expression_label,
+    pn.aes(x="median core expression_pao1", y="median core expression_pa14"),
+)
+fig += pn.geom_point(pn.aes(color="Strain type_pao1"), alpha=0.2)
+fig += pn.labs(
+    x="median expression of core genes (TPM)",
+    y="median expression of core genes (TPM)",
+    title="TPM of core genes",
+)
+fig += pn.theme_bw()
+fig += pn.theme(
+    legend_title_align="center",
+    plot_background=pn.element_rect(fill="white"),
+    legend_key=pn.element_rect(fill="white", colour="white"),
+    legend_title=pn.element_text(family="sans-serif", size=15),
+    legend_text=pn.element_text(family="sans-serif", size=12),
+    plot_title=pn.element_text(family="sans-serif", size=15),
+    axis_text=pn.element_text(family="sans-serif", size=10),
+    axis_title=pn.element_text(family="sans-serif", size=12),
+)
+fig += pn.guides(colour=pn.guide_legend(override_aes={"alpha": 1}))
+
+print(fig)
+
+# +
+# Plot log-scaled
+fig2 = pn.ggplot(
+    pao1_pa14_core_expression_label,
+    pn.aes(x="median core expression_pao1", y="median core expression_pa14"),
+)
+fig2 += pn.scales.scale_x_log10()
+fig2 += pn.scales.scale_y_log10()
+fig2 += pn.geom_point(pn.aes(color="Strain type_pao1"), alpha=0.4)
+fig2 += pn.labs(
+    x="median expression of core genes (log10 TPM)",
+    y="median expression of core genes (log10 TPM)",
+    title="TPM of core genes",
+)
+fig2 += pn.theme_bw()
+fig2 += pn.theme(
+    legend_title_align="center",
+    plot_background=pn.element_rect(fill="white"),
+    legend_key=pn.element_rect(fill="white", colour="white"),
+    legend_title=pn.element_text(family="sans-serif", size=15),
+    legend_text=pn.element_text(family="sans-serif", size=12),
+    plot_title=pn.element_text(family="sans-serif", size=15),
+    axis_text=pn.element_text(family="sans-serif", size=10),
+    axis_title=pn.element_text(family="sans-serif", size=12),
+)
+fig2 += pn.guides(colour=pn.guide_legend(override_aes={"alpha": 1}))
+
+
+print(fig2)
+# -
+
+# **Takeaway:**
+# * This is a positive control that core genes are expressed in both PAO1 and PA14 samples
+#
+# * Based on expression profiles of core genes in PAO1 vs PA14 samples, there doesn't look to be a clear clustering amongst PAO1 and PA14 samples. Is this expected?
```

Comment:
possibly, if the PCA is being driven by outlier expression after rescaling the data it could be that you see clusters. I would at least expect to see some structure... but I'm not very familiar with non-mammalian gene expression.

Reply:
Looks like if I normalize and use UMAP to plot core gene expression of samples, there is some clustering of PAO1 vs PA14, which I would expect. This clustering isn't seen using PCs though, so I guess that means that more dimensions are needed for this separation.